Sources/Receipt/ReceiptDetailSectionData.swift

```diff
@@ -103,13 +103,23 @@ struct ReceiptDetailSectionFeeData: ReceiptDetailSectionData {
             let transaction: Double = receipt.entry == .debit
                 ? 0 - amount - fee
                 : amount - fee
+            let transactionFormat = getTransactionFormat(basedOn: receipt.amount)
             rows.append((title: "receipt_details_transaction".localized(),
                          value: String(format: valueCurrencyFormat,
-                                       String(format: "%.2f", transaction ),
+                                       String(format: transactionFormat, transaction),
                                        receipt.currency)
             ))
         }
     }
+
+    private func getTransactionFormat(basedOn value: String) -> String {
+        let locale = Locale(identifier: Locale.preferredLanguages[0])
```

Comment:
The locale should not be based on the system language, instead, the locale should be base on the receipt currency. There is a discussion going on with @fmattos-hw , we can merge this pr first and revisit it later